src/php/Compiler/Analyzer/Environment/GlobalEnvironment.php

```diff
@@ -32,6 +34,12 @@ final class GlobalEnvironment implements GlobalEnvironmentInterface
 
     private bool $allowPrivateAccess = false;
 
+    public function __construct()
+    {
+        // Add internal definition
+        $this->addInternalDefinition('phel\core', Symbol::create('*compile-mode*'), false);
```

Comment:
Is this comment relevant? I mean, the function's name is already revealing that intention, no?

Reply:
Removed the comment